### ISSUE TITLE
UpgradeProject is still offered even if the suggested language version is unsupported by the target framework

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#47983]
Related to #47977

Repro steps in VS 16.8-preview3:

1. Create a new .NET Core console app
2. Verify that TargetFramework is `netcoreapp3.1`, edit if necessary
3. Replace Program.cs contents with `System.Console.WriteLine("Hello");`
4. Hit ctrl+.

Expected: no UpgradeProject fix is offered, since C# 9 is not supported with netcoreapp3.1
Actual: UpgradeProject fix is offered

/cc @jasonmalinowski @ryzngard

**Comment by davkean** [comment]
This change https://github.com/dotnet/roslyn/pull/44911/files regressed it for. This changed the property from `MaxSupportedLangVersion` to `_MaxSupportedLangVersion`, so this is no longer passed through to Roslyn.

**Comment by sharwell** [comment]
@dotnet/roslyn-compiler I moved this to compiler area path since it's a regression in the build targets.

**Comment by RikkiGibson** [comment]
Tagging @chsienki

**Comment by chsienki** [comment]
Hmm, so we renamed this because it was never supposed to be public. Evidently we had people relying on it without us knowing. 

We also broke xamarin when changing this https://github.com/xamarin/xamarin-android/issues/5049 but they just recently included a fix to update it. 

From the project system it looks like perhaps this is being passed explicitly? https://github.com/dotnet/project-system/pull/5028/files so is a fix just to update that property?

If not I'm leaning towards simply adding `<MaxSupportedLangVersion>$(_MaxSupportedLangVersion)</MaxSupportedLangVersion>` so that we would base all the logic off of the 'private' property, but still expose the result 'publicly'.
